src/routes/categories/containers/CategoriesPage.js

```diff
@@ -1,107 +1,150 @@
+   * else we just dispatch the language change urls here
+   */
   componentDidMount () {
+    if (this.props.categoryId) {
+      const category = this.props.categories.getCategoryById(this.props.categoryId)
+      if (!category) {
+        // todo show an error that the page has not been found, redirect to the notFound route
+      }
+      this.props.replaceUrl(category.url)
+    }
     this.setLanguageChangeUrls(this.props.path)
   }
 
-  // we must not call dispatch in componentWillUpdate or componentDidUpdate
+  /**
+   * Dispatches the action to set the language change urls after a prop change (selection of a category)
+   * we must NOT call dispatch in componentWillUpdate or componentDidUpdate
+   * @see https://reactjs.org/docs/react-component.html#componentwillupdate
+   * @param nextProps The new props
+   */
   componentWillReceiveProps (nextProps) {
     if (nextProps.path !== this.props.path) {
       this.setLanguageChangeUrls(nextProps.path)
     }
   }
 
+  /**
+   * The function used to map different languages to their CategoriesPage
+   * @param {string} language The language
+   * @param {string | undefined} id The id of a category
+   * @returns {string} The url of the CategoriesPage of a different language
+   */
   mapLanguageToUrl = (language, id) => (
-    id ? `/${this.props.location}/${language}/redirect?id=${id}` : `/${this.props.location}/${language}`
+    id ? `/${this.props.location}/${language}?id=${id}` : `/${this.props.location}/${language}`
   )
 
   /**
    * Gets and stores the available languages for the current page
    * @param {string} path The current path
    */
   setLanguageChangeUrls (path) {
-    const hierarchy = new Hierarchy(path)
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      // todo handle this error
-      return
-    }
-    this.props.dispatch(setLanguageChangeUrls(
-      this.mapLanguageToUrl, this.props.languages, hierarchy.top().availableLanguages)
+    this.props.setLanguageChangeUrls(
+      this.mapLanguageToUrl, this.props.languages, this.props.categories.getCategoryByUrl(path).availableLanguages
     )
   }
 
   getPdfFetchPath () {
-    let path = `/${this.props.location}/${this.props.language}/fetch-pdf/`
-    if (this.props.path) {
-      path += this.props.path
-    }
-    return path
+    return `/${this.props.location}/${this.props.language}/fetch-pdf?url=${this.props.path}`
   }
 
-  render () {
-    const url = this.getParentPath()
-    const hierarchy = new Hierarchy(this.props.path)
+  /**
+   * Returns the content to be displayed, based on the current category, which is
+   * a) page with information
+   * b) table with categories
+   * c) list with categories
+   * @param category The current category
+   * @return {*} The content to be displayed
+   */
+  getContent (category) {
+    if (!category) {
+      // todo show an error that the page has not been found, redirect to the notFound route
+      return
+    }
```

Comment:
You can remove this check, if you apply my suggestion below.

Reply:
we just want to display an error, not redirect to a not found page? because disclaimer, language change and so on don't make sense if no category is found, so imo it would be better to redirect to a not found page, maybe we should do this in WEBAPP-90

Reply:
Hmm.. I think the disclaimer isn't affected, if a page is not found in a specific location. Also Language change should default back to the home of the location, if theres no Id to go to. I think we shouldn't redirect to some other url, but just show the error at the page where the error occurs.

Reply:
okay, done